config/module.go

```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/pkg/errors"
 )
 
-var moduleNameRegEx = regexp.MustCompile(`^[A-Za-z0-9_-]+$`)
+var moduleNameRegEx = regexp.MustCompile(`^([a-z0-9-]+:)?[\w-]+$`)
```

Comment:
This change may be controversial but we need to figure out what we are doing with names

Reply:
Looked back at what we discussed in project channel:

```
Rules for public_namespace
[a-z0-9-]+
Rules for module name
[a-z-A-Z-0-9-_]+
```

So this makes sense to me. And it also supports both orgID:name and namespace:name. Are you saying its controversial because we are doing this validation on the backend as well?

Reply:
This is only controversial because it is a non-trivial change to what we are allowing as a module `name`

Reply:
Right right I see. For existing module configs too. Hmm. We can either:
1. See how big of a problem this is for existing configs and do a data migration if this would break something.
2. Only do the validation when type=registry

Reply:
> Only do the validation when type=registry

Rdk modules have no idea that the `Type` field exists on app

Reply:
Discussed offline: https://viam.atlassian.net/browse/APP-2430

Reply:
Agreed. Hotfix to replace : with _ works for now. Followup can do the actual-right-thing in another PR.